src/Engine/ProtoCore/CodeGen.cs

```diff
@@ -91,7 +91,7 @@ public abstract class CodeGen
 
 
         // Contains the list of Nodes in an identifier list
-        protected List<ProtoCore.AST.AssociativeAST.AssociativeNode> ssaPointerList;
+        protected Stack<List<ProtoCore.AST.AssociativeAST.AssociativeNode>> stackSSAPointerList;
```

Comment:
sorry for nitpicking, as it is a stack, probably removing `List` from this variable name would make the code more readable...